bodhi/tests/server/test_buildsys.py

```diff
@@ -140,3 +141,53 @@ def test_login_success(self, error):
         self.assertEqual(type(client), koji.ClientSession)
         # No error should have been logged
         self.assertEqual(error.call_count, 0)
+
+
+class TestGetSession(unittest.TestCase):
+    """Tests :func:`bodhi.server.buildsys.get_session` function"""
+
+    @mock.patch('bodhi.server.buildsys._buildsystem', None)
+    def test_uninitialized_buildsystem(self):
+        """Assert calls to get_session raise RuntimeError when uninitialized"""
+        self.assertRaises(RuntimeError, buildsys.get_session)
+
+    @mock.patch('bodhi.server.buildsys._buildsystem')
+    @mock.patch('bodhi.server.buildsys._buildsystem_login_lock', wraps=Lock())
+    def test_buildsys_lock(self, mock_lock, mock_buildsystem):
+        """Assert the buildsystem lock is aquired and released in get_session"""
+        session = buildsys.get_session()
+        mock_lock.__enter__.assert_called_once()
+        mock_lock.__exit__.assert_called_once()
+        mock_buildsystem.assert_called_once_with()
+
+
+class TestSetupBuildsystem(unittest.TestCase):
+    """Tests :func:`bodhi.server.buildsys.setup_buildsystem` function"""
+
+    @mock.patch('bodhi.server.buildsys._buildsystem', mock.Mock())
+    def test_initialized_buildsystem(self):
+        """Assert nothing happens when the buildsystem is already initialized"""
+        old_buildsystem = buildsys._buildsystem
+        buildsys.setup_buildsystem({})
+        self.assertTrue(old_buildsystem is buildsys._buildsystem)
+
+    @mock.patch('bodhi.server.buildsys._buildsystem', None)
+    def test_koji_buildsystem(self):
+        """Assert the buildsystem initializes correctly for koji"""
+        self.assertTrue(buildsys._buildsystem is None)
+        buildsys.setup_buildsystem({'buildsystem': 'koji'})
+        self.assertFalse(buildsys._buildsystem is None)
```

Comment:
We could probably add some stronger assertions here by calling the buildystem (since it's that weird internal function) and making assertions on the return value. Perhaps asserting that the type of the buildsystem is a KojiSomethingOrOther, and that it got the config settings? You could just set up a mock on ```koji_login()``` and assert that it gets called when you call the buildsystem and assert that it gets the right settings.